operators/tokenizer/bpe_tokenizer.hpp

```diff
@@ -54,7 +55,29 @@ inline bool IsUnicodeSpace(char32_t ch) {
 }
 
 inline bool IsEmptyUString(const ustring& str) {
-  return std::all_of(str.begin(), str.end(), [](char32_t ch) { return IsUnicodeSpace(ch); });
+  if (str == ustring(" ")) {
+    return false;
+  } else {
+    return std::all_of(str.begin(), str.end(), [](char32_t ch) { return IsUnicodeSpace(ch); });
+  }
+}
+
+inline bool BothSpaces(char32_t lhs, char32_t rhs) {
+    return (lhs == rhs) && (lhs == ' ');
+}
+
+inline ustring ReplaceString(ustring str, const ustring& search, const ustring& replace) {
+    size_t pos = 0;
+    while ((pos = str.find(search, pos)) != ustring::npos) {
+    str.replace(pos, search.length(), replace);
+        pos += replace.length();
+    }
+    return str;
+}
+
+inline void WhiteSpaceClean(ustring& str) {
+    str = ReplaceString(str, ustring("\n"), ustring(" "));
+    str.erase(std::unique(str.begin(), str.end(), BothSpaces), str.end());
```

Comment:
I don't remember str.erase can remove multiple iterators, can you confirm this line can remove multiple occurrences of WS sequences